examples/simple_example/main.tf

```diff
@@ -0,0 +1,73 @@
+module "iap_bastion" {
+  source  = "../.."
+  project = var.project
+  zone    = var.zone
+  network = google_compute_network.network.self_link
+  subnet  = google_compute_subnetwork.subnet.self_link
+  members = var.members
+}
+
+# Address for NATing
+resource "google_compute_address" "nat" {
+  project = var.project
+  region  = var.region
+  name    = "bastion-nat-external"
+}
+
+# Create a NAT router so the nodes can reach the public Internet
+resource "google_compute_router" "router" {
+  name    = "bastion-router"
+  project = var.project
+  region  = var.region
+  network = google_compute_network.network.self_link
+  bgp {
+    asn = 64514
+  }
+}
+
+# NAT on the main subnetwork
```

Comment:
Should NAT be incorporated into the module?

Can we replace w/ https://github.com/terraform-google-modules/terraform-google-cloud-nat ?

Reply:
Great suggestion, I think that should probably be in the module as an option. Especially with private google access, many customers may not want internet egress on a bastion, but it's worth giving them a flag to allow it.

Reply:
I'm a bit confused: if the suggestion is to reference the NAT module from this module (instead of using it in the example), I'm puzzled by the perceived advantage. People able to reference this module in their Terraform setups, can as well be able to reference the NAT module next to it. Adding it in the module only generates unnecessary complexity and hides important infrastructure components behind an option.

Reply:
After some discussion, I'm going to remove the NAT code from the simple example. It's a nice to have for downloading packages in a startup script for testing, but in practice, this should be done as part of an image bakery. I agree @ludoo that adding it in is unnecessary.